#139456249  setting up test enviroment and integrating coverall code coverage service

freemile2ru opened the pull request:
# **What does this PR do?**

    Setup unit testing for the front end
    Setup coverall code coverage service

# **Description of Task to be completed?**
   1. this PR sets jasmine framework to test our codes and also configures karma to make firefox, chrome and phantom js our testing environment
   2. it also configures karma to send our test coverage result to travis using covaralls.  

# **How should this be manually tested?**

## **for the unit testing**
(1)clone this repository
(2)create the .js file you want to test and save.
(3) enter the test folder
(4) create new .js file you want to write your test in
(5) include the scripts path in the specRunner.html file with the actual code path coming before the test code path
(6)load the specRunner.html file in your browser
   

## For test coverage service
 (1) clone the repo,
 (2) install all dependencies
 (3) navigate to the location on your Cli
 (4) run karma start karma2.conf.js
 (5) the browsers will be launched to show karma is working and the tests results on each browser will be displayed on the console
 (6) the test coverage result will be shown the coverage folder and on this repository when pushed.

# **Any background context you want to provide?**
  this PR contains a travis.yml file for your test coverage

# **What are the relevant pivotal tracker stories?**
    Setup unit testing for the front end
    Setup coverall code coverage service
    
Screenshots (if appropriate)
Questions:

Commit:
chore(testing): successfully setup unit testing for the front end

Commit:
chore(.gitignore): adding coverage folder to .gitignore file

Commit:
chore(.gitignore): adding coverage folder to .gitignore file#2

Commit:
chore(coveralls): coveralls code coverage service successfully integrated

Commit:
chore(travis): modified travis.yml file

Commit:
chore(travis): modified test spec file

ooduye commented on lib/jasmine-2.3.4/boot.js:
```diff
@@ -0,0 +1,121 @@
+/**
+ Starting with version 2.0, this file "boots" Jasmine, performing all of the necessary initialization before executing the loaded environment and all of a project's specs. This file should be loaded after `jasmine.js` and `jasmine_html.js`, but before any project source files or spec files are loaded. Thus this file can also be used to customize Jasmine for a project.
```
Remove the lib/ directory from the pull request

ooduye commented on config/env/all.js:
```diff
@@ -5,5 +5,5 @@ var keys = rootPath + '/keys.txt';
 	root: rootPath,
 	port: process.env.PORT || 3000,
-    db: process.env.MONGOHQ_URL
+    db: `mongodb://freemile:Kratus043@ds151279.mlab.com:51279/card4human`
```
Change this to reflect just the environment variable

Commit:
chore(lib folder): remove lib folder and set env variables

Commit:
chore(env): added .env file

Commit:
chore(env): added .env file

Commit:
chore(env): added .env file

Commit:
Merge branch 'develop' into chore/#139456249/integrating-coverall-code-coverage-service

ooduye commented:
Your travis-ci build is failing. Please check it and find out why.

Commit:
bug(travis build): fixed travis build fail error

Commit:
bug(travis build): fixed travis build fail error #2

Commit:
bug(travis build): fixed travis build fail error #3

Commit:
bug(travis build): fixing travis build fail error #4

freemile2ru commented:
hello oluwayemisi,
  i have fixed the bug and travis is now building. thanks

Commit:
chore(fileName): renamed files properly

Commit:
chore(linters): renamed files (from andela.js to test.js) [finish #139456249]

Commit:
chore(test coverage): renamed files (from andela.js to test.js) [finish #139456249]

Commit:
Merge branch 'develop' into chore/#139456249/integrating-coverall-code-coverage-service

Commit:
bug(bug fixed): (fixed travis build error) [finish #139456249]

ooduye commented:
@andela-uibrahim Your builds are failing...

freemile2ru commented:
hello @andela-ooduye,
 I have implemented the feedback you requested. i have changed the file names, change the coverage  file format and the travis is now building. please help review the current status
